Query Builder - Fix UI for long queries / get rid of overlap

kristynrb opened the pull request:
I'm attempting to fix the UI problem where the View and Grouped By sections aren't formatted responsively.  

I'd like to put scroll functionality, but noticed in another issue that someone mentioned the original intention may be to stack these. If stacking is the intention, does anyone have a wireframe or suggestion on what the UI should look like when there are longer aggregations? 

If you're open to adding a scroll, is there a suggestion on which scroll to use (I noticed a few scroll packages were used throughout the code and these section aren't setup in the same way as the Filtered By section that currently has scroll functionality). 

Happy to continue working on this with some guidance!

This will also help to solve [issue 5325](https://github.com/metabase/metabase/issues/5325)
![fix formatting](https://i.imgur.com/w5NjTRJ.png).

###### TODO 
-  [x] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
(unless it's a tiny documentation change).

kristynrb committed:
initial installation

kristynrb committed:
updated layout for agg list and padding for add button

salsakran commented:
thank you for the help @kristynrb !

@kdoh or @attekei mind taking a look at this? I think our original intent was for these to have scrollbars, not sure if they got dropped on a regression somewhere .

mazameli commented:
Thanks for the nudge; I forgot I was going to comment on this. https://github.com/metabase/metabase/issues/5470 is the issue that points out where we lost the scrollbars for the filter area on some browsers. I don't recall if the View or Groupings sections ever had scroll bars, but I think it would make sense for them to use the same solution that we use for the filters area.

kdoh commented on frontend/src/metabase/query_builder/components/GuiQueryEditor.jsx:
```diff
@@ -212,7 +212,11 @@ export default class GuiQueryEditor extends Component {
             }
-            return aggregationList
+            return (
+              <div className={cx("Query-section Query-section-breakout scroll-x scroll-show scroll-show--horizontal")}>
```
I'm not sure we need the `Query-section-breakout` class here since this is the aggregation list.

kristynrb replied:
@kdoh I had put that in there to give the aggregations a parent div and was was playing around with trying to get the scroll on this parent (which affected both the query section and the aggregation). I understand that it's misnamed and I can adjust that. I can remove the name and merge the changes from master into this before continuing to work on it. Thanks!

kdoh reviewed (changes requested):
Left a small comment about the `Query-section-breakout` class. While I was poking at this I made a little progress on figuring out why the scrollbars aren't visible despite the use of `scroll-show.` I think the right move would be for me to cut a quick PR to fix that and then when that hits master just merge that into this and then things should work as intended.

kdoh commented:
Hey there @kristynrb. My scroll fix change just hit master. If you merge master or rebase your branch to pick up those changes you'll see the scroll bars for the relevant sections in this PR.

Happy to re-review and merge this PR once you make that class name change.

kristynrb committed:
removed class name

kristynrb committed:
Merge branch 'master' of https://github.com/metabase/metabase into long_filter_issue

kristynrb committed:
wip - merged master w scroll changes. Need to finish fixing scroll on other aggs

kdoh commented:
Hey @kristynrb. Noticed you've added some commits to this. Should I take another look yet?

kristynrb commented:
Hi @kdoh - no, the changes aren't ready yet. Sorry - I started on them one day and didn't complete them. I'll try to get back to it this afternoon or Monday and will reach out when it's ready for review. Thank you!

kristynrb committed:
Merge branch 'master' of https://github.com/metabase/metabase into long_filter_issue

kristynrb committed:
cleaning up AggregationList

kristynrb committed:
updated breakout filter

kristynrb committed:
updated filter and agg with scroll

kristynrb commented:
@kdoh Thanks again for updating your code! I've added the classes in to the view. Feel free to take a look, because I'd love to hear your suggestions on a few things. 

![qb view](https://i.imgur.com/oZGfQD3.png)

Unfortunately, each of these sections are setup differently, so I seem to be running into different issues with each, but nothing we can't overcome.

1. The `and` for the aggregations has gone wonky when I added the scroll.
2. I've been trying to move the `+` boxes so that they're not a part of the same element as the lists for filter and aggregation - that way they can be permanently to the right and not affected by the scroll. Unfortunately, they're built into the actual widgets. This may take some big finagling unless you're seeing something that I don't see.
3. Responsiveness is not very friendly on this whole view. It should probably be its own issue (if it's not already). As for now, I've just set some max-widths on the lists. I think that we'll need to work setting up the parents a little bit better before these children can be appropriately responsive.

tschut commented:
Hi, are there any plans to have another look at this? @kristynr are you planning to fix the merge conflicts? 

I see this issue isn't scheduled for any milestone, it would really benefit some people here. A workaround is to zoom out to a ridiculous level in the browser.

kristynrb commented:
Hi - sorry, I'm no longer working on this project.

On Wed, Jul 11, 2018 at 5:50 AM Tiemen Schut <notifications@github.com>
wrote:

> Hi, are there any plans to have another look at this? @kristynr are you
> planning to fix the merge conflicts?
>
> I see this issue isn't scheduled for any milestone, it would really
> benefit some people here. A workaround is to zoom out to a ridiculous level
> in the browser.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/metabase/metabase/pull/6811#issuecomment-404157159>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALnWb7dKTWkXmNcANyA3VryDH1ua_frrks5uFfSFgaJpZM4RxGwb>
> .
>
